### DISCUSSION TITLE
How to get the Distinguished Names (DNs) in the certificate?

### DISCUSSION THREAD

**Discussion by rajesh1084** [smallstep/certificates#1407] [Q&A] [answered]
Tried to embed the DNs in the subject something like -
```
export DN="C=AU,ST=Victoria,L=Melbourne,O=Digicert,OU=Sales,CN=www.knowledge.com"
step certificate create $DN test.crt test.key
```
Also tried providing the details using the flag `--set key=value` -
```
step certificate create ... \
 --set countryName=AU \
 --set stateOrProvinceName=Victoria \
 --set localityName=Melbourne \
...
```

However, I don't see this information while inspecting the received certificate. Any pointers?

**Comment by hslatman** [comment, marked as answer]
Hey @rajesh1084, you will need to configure a template to accomplish this. A user with a similar use case asked their question [here](https://github.com/smallstep/certificates/discussions/662).

If the organizational information in your DN is static and the same for each certificate, you can configure it on the authority level. Otherwise, you can do so on the provisioner level. It's also possible to make the user provided values, but you'll need to change the default template to do so.

Our documentation about templates can be found [here](https://smallstep.com/docs/step-ca/templates/).

**Comment by maraino** [comment]
These values can be set up in two different ways: 
1. At the provisioner level in a  template. They can be fixed values, they can come from the user like `--set foo=bar`, or they can come from a webhook. Like the first comment in https://github.com/smallstep/certificates/discussions/662#discussioncomment-1121511
2. At the authority level, all provisioners will use these values unless they are overwritten at the provisioner level. Like the second comment https://github.com/smallstep/certificates/discussions/662#discussioncomment-1122023
